### PR TITLE
feat: show voucher rewards in transaction details

### PR DESCRIPTION
**Pull request by peterjah**

### CODE REVIEW

**Comment by dlecan** [comment]
https://spacelephant.tpondemand.com/entity/2716

File: src/components/transaction/Details.vue [hunk 1]
```diff
@@ -60,8 +60,34 @@
...
+              trigger: 'hover click',
+              content: price ? readableCurrency(rewards.sender, price) : '',
+              placement: 'left',
+            }"
+            >{{ readableCrypto(rewards.sender) }}
+            </div>
+          </div>
+
+          <div class="list-row-border-b">
+            <LinkUNIK :id="foundationUnikid" :unikname="$t('TRANSACTION.REWARDS.FOUNDATION')" :type='"ORGANIZATION"'></LinkUNIK>
```
Comment (ftreguer) [inline src/components/transaction/Details.vue]: you can directly use type="ORGANIZATION" `:` is for interpretation

File: src/components/transaction/Details.vue [hunk 2]
```diff
@@ -357,6 +419,29 @@ export default class TransactionDetails extends Vue {
...
+    while (
+        milestoneIdx < config.milestones.length - 1 &&
+        height >= config.milestones[milestoneIdx + 1].height
+    ) {
+        milestoneIdx++;
+    }
+
+    while (milestoneIdx && height < config.milestones[milestoneIdx].height) {
+        milestoneIdx--;
+    }
```
Comment (ftreguer) [inline src/components/transaction/Details.vue]: Not very readable :smile: Comment?

File: src/mixins/transaction-types.ts [hunk 3]
```diff
@@ -113,6 +114,9 @@ export default {
     isUnsCertifiedNftMint(type: number, typeGroup: number): boolean {
       return isUNSTypeGroup(typeGroup) && type === UNSTransaction.CERTIFIED_NFT_MINT;
     },
+    isVoucherUnsCertifiedNftMint(transaction: ITransaction): boolean {
+      return isUNSTypeGroup(transaction.typeGroup) && transaction.type === UNSTransaction.CERTIFIED_NFT_MINT && transaction.asset.nft.unik.properties.UnikVoucherId;
+    },
```
Comment (ftreguer) [inline src/mixins/transaction-types.ts]: Can you use `isUnsCertifiedNftMint` function in `isVoucherUnsCertifiedNftMint`?

File: src/App.vue [hunk 4]
```diff
@@ -92,7 +92,14 @@ export default class App extends Vue {
       this.$store.dispatch("network/setKnownWallets", knownWallets);
     }
 
-    this.fetchInitialSupply();
+    const crypto = await NodeService.crypto();
+    this.$store.dispatch("network/setNetworkConfig", crypto);
```
Comment (ftreguer) [inline src/App.vue]: You can call `updateNetworkConfig` here

File: src/App.vue [hunk 5]
```diff
@@ -92,7 +92,14 @@ export default class App extends Vue {
...
+
+    let initialSupply = localStorage.getItem("initialSupply");
+    if (!initialSupply) {
+      initialSupply = crypto.genesisBlock.totalAmount;
+    }
+    this.$store.dispatch("network/setInitialSupply", initialSupply);
```
Comment (ftreguer) [inline src/App.vue]: An old merge mistake? :+1:

**Review by ftreguer — CHANGES_REQUESTED** [review]

File: src/components/transaction/Details.vue (hunk 1 above)
Comment (peterjah) [reply src/components/transaction/Details.vue]: no !! its not working without simple quotes

File: src/components/transaction/Details.vue (hunk 2 above)
Comment (peterjah) [reply src/components/transaction/Details.vue]: agree... its code from ark-crypto but i don't see the point....

File: src/components/transaction/Details.vue (hunk 1 above)
Comment (ftreguer) [reply src/components/transaction/Details.vue]: it works `<LinkUNIK :id="foundationUnikid" :unikname="$t('TRANSACTION.REWARDS.FOUNDATION')" type="ORGANIZATION"></LinkUNIK>`

**Review by ftreguer — APPROVED** [review]

**Pull request merged**